examples/mnist/run.py

```diff
@@ -131,22 +137,22 @@ def receive_output(self, likelihoods: tf.Tensor) -> tf.Operation:
 with tfe.protocol.Pond(server0, server1, crypto_producer) as prot:
 
     # get model parameters as private tensors from model owner
-    w0, b0, w1, b1 = prot.define_private_input(model_trainer, masked=True)  # pylint: disable=E0632
+    params = prot.define_private_input(model_trainer.player, model_trainer.provide_input, masked=True)  # pylint: disable=E0632
 
     # we'll use the same parameters for each prediction so we cache them to avoid re-training each time
-    w0, b0, w1, b1 = prot.cache([w0, b0, w1, b1])
+    params = prot.cache(params)
 
     # get prediction input from client
-    x = prot.define_private_input(prediction_client, masked=True)  # pylint: disable=E0632
+    x, = prot.define_private_input(prediction_client.player, prediction_client.provide_input, masked=True)  # pylint: disable=E0632
 
     # compute prediction
+    w0, b0, w1, b1 = params
     layer0 = prot.matmul(x, w0) + b0
     layer1 = prot.sigmoid(layer0 * 0.1)  # input normalized to avoid large values
-    layer2 = prot.matmul(layer1, w1) + b1
-    prediction = layer2
+    logits = prot.matmul(layer1, w1) + b1
 
     # send prediction output back to client
-    prediction_op = prot.define_output(prediction, prediction_client)
+    prediction_op = prot.define_output(prediction_client.player, [logits], prediction_client.receive_output)
```

Comment:
should be `logits` instead of `[logits]`, no?

Reply:
Both are allowed. I can change it if you prefer?